### PR TITLE
Add method to test if line segment intersects a convex polygon.

### PR DESCRIPTION
**Pull request by dnesbitt61**
Cleanup comments and formatting too

### CODE REVIEW

File: src/midgard/linesegment2.cc [hunk 1]
```diff
@@ -1,93 +1,212 @@
...
-      closest = a_;
+// Finds the distance squared of a specified point from the line segment
+// and the closest point on the segment to the specified point.
+float LineSegment2::DistanceSquared(const Point2& p, Point2& closest) const {
+  // Construct vector v (ab) and w (ap)
+  Vector2 v(a_, b_);
+  Vector2 w(a_, p);
+
+  // Numerator of the component of w onto v. If <= 0 then a
+  // is the closest point. By separat_ing into the numerator
```
Comment (gknisely) [inline src/midgard/linesegment2.cc]: typo in comment-->   separat_ing

File: src/midgard/linesegment2.cc [hunk 2]
```diff
@@ -1,93 +1,212 @@
...
+  return sqrtf(DistanceSquared(p, closest));
+}
 
-  bool LineSegment2::Intersect(const LineSegment2& segment,
-               Point2& intersect) const {
-    // Construct vectors
-    Vector2 b = b_ - a_;
-    Vector2 d = segment.b() - segment.a();
+// Determines if the current segment intersects the specified segment.
+// If an intersect occurs the intersectPt is determined.  Note: the
```
Comment (gknisely) [inline src/midgard/linesegment2.cc]: same--> intersectPt

File: src/midgard/linesegment2.cc [hunk 3]
```diff
@@ -1,93 +1,212 @@
...
+    // Set an outward facing normal (polygon is assumed to be CCW)
+    Vector2 n(pt2->y() - pt1->y(), pt1->x() - pt2->x());
+
+    // Dot product of the normal to this polygon edge with the ray
+    float n_dot_c = n.Dot(c);
+    float num = n.Dot((*pt1 - a_));
+
+    // Check for parallel line
+    if (std::abs(n_dot_c) < kEpsilon) {
+      // No intersection if segment origin is outside wrt to this edge
```
Comment (gknisely) [inline src/midgard/linesegment2.cc]: wrt?

Comment (kevinkreiser) [inline src/midgard/linesegment2.cc]: yeah need to remove that extra to in there as well i guess haha

**Pull request merged**